src/MusicWheel.cpp

```diff
@@ -600,7 +602,21 @@ void MusicWheel::BuildWheelItemDatas( std::vector<MusicWheelItemData *> &arrayWh
 					bUseSections = false;
 					break;
 				case SORT_TOP_GRADES:
-					SongUtil::SortSongPointerArrayByGrades( arraySongs, true );
+						SongUtil::SortSongPointerArrayByGrades( arraySongs, true );
+					break;
+				case SORT_TOP_GRADES_P1:
+					// Check if master player profile is persistent
+					if( PROFILEMAN->IsPersistentProfile(PLAYER_1) )
+						SongUtil::SortSongPointerArrayByProfileGrades( arraySongs, true, PLAYER_1);
+					else
+						SongUtil::SortSongPointerArrayByGrades( arraySongs, true );

```

Comment:
My gut feeling is that we should maybe return empty instead of the same as SORT_TOP_GRADES. In the theme I think the option wouldn't be even displayed if it wasn't a persistent profile but engine side probably empty is more correct if P1/P2 data isn't there.

Reply:
Yeah i think you're right. I'll change it here